README.md
# Docker Pydio

[![Version](https://images.microbadger.com/badges/version/cedvan/pydio:8.0.1-2.svg)](https://microbadger.com/images/cedvan/pydio:8.0.1-2 "Get your own version badge on microbadger.com")
[![Docker Pulls](https://img.shields.io/docker/pulls/cedvan/pydio.svg?style=flat-square)](https://hub.docker.com/r/cedvan/pydio/)
[![Docker Stars](https://img.shields.io/docker/stars/cedvan/pydio.svg?style=flat-square)](https://hub.docker.com/r/cedvan/pydio/)
[![Docker image](https://images.microbadger.com/badges/image/cedvan/pydio.svg?style=flat-square)](https://microbadger.com/#/images/cedvan/pydio "Size docker image on Docker Hub")

[![Build Status](https://img.shields.io/travis/cedvan/docker-pydio/master.svg?style=flat-square)](https://travis-ci.org/cedvan/docker-pydio)
[![Average time to resolve an issue](http://isitmaintained.com/badge/resolution/cedvan/docker-pydio.svg)](http://isitmaintained.com/project/cedvan/docker-pydio "Average time to resolve an issue")
[![Percentage of issues still open](http://isitmaintained.com/badge/open/cedvan/docker-pydio.svg)](http://isitmaintained.com/project/cedvan/docker-pydio "Percentage of issues still open")
[![Join the chat at https://gitter.im/cedvan/docker-pydio](https://badges.gitter.im/cedvan/docker-pydio.svg)](https://gitter.im/cedvan/docker-pydio?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
[![GitHub license](https://img.shields.io/:license-mit-blue.svg?style=flat-square)]()

![pydio](https://raw.githubusercontent.com/cedvan/docker-pydio/master/img/pydio.png "pydio")

Complete Solution of sharing and sync for your employees. Open source software deployed on your servers or in your cloud.

## Quick start

```bash
docker run --name pydio -d \
    -p 80:80 \
    cedvan/pydio:8.0.1-2
```
- Go with your browser to **localhost**
- Install pydio with wizard
- Configure database with embedded MySQL server, cf [Use MySQL embedded](#use-mysql-embedded)

## Configure specific DNS

To use a specfic DNS, configure pydio host.

```bash
docker run --name pydio -d \
    -p 80:80 \
    -e "PYDIO_HOST=domain.com" \
    cedvan/pydio:8.0.1-2
```

## Save Pydio data

Pydio files are saved to `/data/pydio` in container. Just mount this volume for save your configurations and files

```bash
docker run --name pydio -d \
    -v /opt/pydio:/data/pydio \
    cedvan/pydio:8.0.1-2
```

## Pydio database

Pydio need a database to works.

### Use MySQL embedded

By default with `DB_HOST` environment variable set to `localhost` or `127.0.0.1`, this container configure and start a embedded MySQL server.

Database default configuration :
- Name: `pydio`
- User: `pydio`
- Password: `pydio`

You can configure this parameters, cf [Pydio Options](#pydio-options)

*Sample :*

```bash
docker run --name pydio -d \
    -v /opt/pydio-mysql:/var/lib/mysql \
    cedvan/pydio:8.0.1-2
```
*Don't forget mount `/var/lib/mysql` to save MySQL data*

And configure database parameters in Pydio with wizard installation

![pydio](https://raw.githubusercontent.com/cedvan/docker-pydio/master/img/pydio-database-embedded.png "pydio-database-embedded")



### Use external database

*Sample with mysql*

```bash
docker run --name pydio-mysql -d \
    -v /opt/pydio-mysql:/var/lib/mysql \
    -e "MYSQL_ROOT_PASSWORD=my_pydio_db_root_password" \
    -e "MYSQL_DATABASE=my_pydio" \
    -e "MYSQL_USER=my_pydio_db_user" \
    -e "MYSQL_PASSWORD=my_pydio_db_password" \
    mysql:latest
```

```bash
docker run --name pydio -d \
    --link pydio-mysql:mysql
    -e "DB_HOST=mysql" \
    -e "DB_NAME=my_pydio" \
    -e "DB_USER=my_pydio_db_user" \
    -e "DB_PASSWORD=my_pydio_db_password" \
    cedvan/pydio:8.0.1-2
```

And configure database parameters in Pydio with wizard installation

## Enabled HTTPS

```bash
docker run --name pydio -d \
    -p 443:443 \
    -e "PYDIO_HTTPS=true" \
    -e "PYDIO_FORCE_HTTPS=true" \
    -v /opt/pydio/certs:/data/certs \
    cedvan/pydio:8.0.1-2
```
*Add your **pydio.key** and **pydio.crt** in folder **certs**. If `pydio.key` and `pydio.crt` do not exist, the container will create self-signed certificates*

## Load Pydio behind HTTPS Proxy

```bash
docker run --name pydio -d \
    -e "PYDIO_HTTPS_PORT=443" \
    -e "PYDIO_HTTPS_REVERSE_PROXY=true" \
    -v /opt/pydio/certs:/data/certs \
    cedvan/pydio:8.0.1-2
```
*Configure your proxy to redirect on container on port 443. No need certificates in container with this configuration*

## Pydio Options

*Please refer the docker run command options for the `--env-file` flag where you can specify all required environment variables in a single file. This will save you from writing a potentially long docker run command. Alternately you can use fig.*

Below is the complete list of available options that can be used to customize your Pydio installation.

- **PYDIO_HOST**: The hostname of the pydio server. Defaults to `localhost`
- **PYDIO_HTTP_PORT**: The port of the pydio http server. Defaults to `80`
- **PYDIO_HTTPS**: Set to `true` to enable https support, Defaults to `false`. **If you do not use a reverse proxy, do not forget to add the certificates files**
- **PYDIO_HTTPS_PORT**: The port of the pydio https server. Defaults to `443`
- **PYDIO_HTTPS_REVERSE_PROXY**: Set to `true` if you use docker behind a reverse proxy for i.e. ssl termination. This will make pydio use the HTTPS scheme without the need to add certificates. If you do so, make sure to set your reverse proxy to target port 443. Defaults to `false`
- **PYDIO_FORCE_HTTPS**: Set to `true` if you want Pydio automatically redirect to https. Defaults to `false`
- **PYDIO_LOCALE**: Set to configure default locale used by Pydio. Defaults to `en_US`
- **DB_HOST**: Set to configure database host. This image provide a embedded mysql, but you can specify external database with this parameter. Defaults to `localhost`
- **DB_PORT**: Set to configure database port. Defaults to `3306`
- **DB_NAME**: Set to configure database name. Defaults to `pydio`
- **DB_USER**: Set to configure database user. Defaults to `pydio`
- **DB_PASSWORD**: Set to configure database password. Defaults to `pydio`
- **PHP_TIMEZONE**: Configure timezone PHP. Default `Europe/Paris`.

## Pdyio License

Pydio is open source and free to the communauty.
You can add a license to activate enterprise edition, cf [https://pydio.com/en/pricing](https://pydio.com/en/pricing)

## Contributing

See [CONTRIBUTING](CONTRIBUTING.md) file.

## References

Pydio is built by Abstrium

- https://pydio.com
- https://github.com/pydio/pydio-core
- http://www.abstrium.com


Changelog.md
# Changelog

**latest**

**8.0.1-2**
- Configure pydio temporary directory

**8.0.1-1**
- Fix vhosts to catch no existing files, important to support shared files

**8.0.1**
- Add base files


scripts/config.sh
#!/bin/bash

echo "Check environment configuration..."

# Directories
DATA_DIRECTORY=/data
WORK_DIRECTORY=/var/www
ASSETS_DIRECTORY=/assets
SCRIPTS_DIRECTORY=/scripts/pydio

# PHP Configuration
PHP_TIMEZONE=${PHP_TIMEZONE:-Europe/Paris}

# Pydio Configuration
PYDIO_HOST=${PYDIO_HOST:-localhost}
PYDIO_HTTP_PORT=${PYDIO_HTTP_PORT:-80}
PYDIO_HTTPS=${PYDIO_HTTPS:-false}
PYDIO_HTTPS_PORT=${PYDIO_HTTPS_PORT:-443}
PYDIO_HTTPS_REVERSE_PROXY=${PYDIO_HTTPS_REVERSE_PROXY:-false}
PYDIO_FORCE_HTTPS=${PYDIO_FORCE_HTTPS:-false}
PYDIO_LOCALE=${PYDIO_LOCALE:-en_US}
DB_HOST=${DB_HOST:-localhost}
DB_PORT=${DB_PORT:-3306}
DB_NAME=${DB_NAME:-pydio}
DB_USER=${DB_USER:-pydio}
DB_PASSWORD=${DB_PASSWORD:-pydio}

# Checking PHP Timezone
if [[ ! "${PHP_TIMEZONE}" =~ ^[A-Z]{1}[a-z]+/[A-Z]{1}[a-z]+$ ]]; then
    echo "ERROR: "
    echo "  Variable PHP_TIMEZONE \"${PHP_TIMEZONE}\" isn't valid ! (Format accepted : [A-Z]{1}[a-z]+/[A-Z]{1}[a-z]+)"
    exit 1
fi

# Checking Pydio HTTPS Configuration and load Pydio scheme
if [ "${PYDIO_HTTPS}" = "true" ]; then
    PYDIO_SCHEME="https"
elif [ "${PYDIO_HTTPS}" = "false" ]; then
    PYDIO_SCHEME="http"
else
    echo "ERROR: "
    echo "  Variable PYDIO_HTTPS \"${PYDIO_HTTPS}\" isn't valid ! (Values accepted : true/false)"
    exit 1
fi

# Checking HTTP PORT
if [[ ! ("${PYDIO_HTTP_PORT}" =~ ^[1-9]{1}[0-9]{0,4}$) ]] || (("${PYDIO_HTTP_PORT}" > "65535")); then
    echo "ERROR: "
    echo "  Variable PYDIO_HTTP_PORT \"${PYDIO_HTTP_PORT}\" isn't valid ! (Format accepted : integer between 1 and 65535)"
    exit 1
fi

# Checking HTTPS PORT
if [[ ! ("${PYDIO_HTTPS_PORT}" =~ ^[1-9]{1}[0-9]{0,4}$) ]] || (("${PYDIO_HTTPS_PORT}" > "65535")); then
    echo "ERROR: "
    echo "  Variable PYDIO_HTTP_PORT \"${PYDIO_HTTPS_PORT}\" isn't valid ! (Format accepted : integer between 1 and 65535)"
    exit 1
fi

# Checking Pydio HTTPS reserve proxy
if [ "${PYDIO_HTTPS_REVERSE_PROXY}" != "true" ] && [ "${PYDIO_HTTPS_REVERSE_PROXY}" != "false" ]; then
    echo "ERROR: "
    echo "  Variable PYDIO_HTTPS_REVERSE_PROXY \"${PYDIO_HTTPS_REVERSE_PROXY}\" isn't valid ! (Values accepted : true/false)"
    exit 1
fi

# Checking Pydio force HTTPS
if [ "${PYDIO_FORCE_HTTPS}" != "true" ] && [ "${PYDIO_FORCE_HTTPS}" != "false" ]; then
    echo "ERROR: "
    echo "  Variable PYDIO_FORCE_HTTPS \"${PYDIO_FORCE_HTTPS}\" isn't valid ! (Values accepted : true/false)"
    exit 1
fi

# Checking Pydio Locale
if [[ ! "${PYDIO_LOCALE}" =~ ^[a-z]{2}+_[A-Z]{2}+$ ]]; then
    echo "ERROR: "
    echo "  Variable PYDIO_LOCALE \"${PYDIO_LOCALE}\" isn't valid ! (Format accepted : [a-z]{2}+_[A-Z]{2}+)"
    exit 1
fi

# Checking database host
if [[ ! "${DB_HOST}" =~ ^[A-Za-z]+$ ]]; then
    echo "ERROR: "
    echo "  Variable DB_HOST \"${DB_HOST}\" isn't valid ! (Format accepted : [A-Za-z\_\-\.]+)"
    exit 1
fi

# Checking database port
if [[ ! ("${DB_PORT}" =~ ^[1-9]{1}[0-9]{0,4}$) ]] || (("${DB_PORT}" > "65535")); then
    echo "ERROR: "
    echo "  Variable DB_PORT \"${DB_PORT}\" isn't valid ! (Format accepted : integer between 1 and 65535)"
    exit 1
fi

# Checking database name
if [[ ! "${DB_NAME}" =~ ^[A-Za-z\_\-]+$ ]]; then
    echo "ERROR: "
    echo "  Variable DB_NAME \"${DB_NAME}\" isn't valid ! (Format accepted : [A-Za-z\_\-]+)"
    exit 1
fi

# Checking database user
if [[ ! "${DB_USER}" =~ ^[A-Za-z\_\-]+$ ]]; then
    echo "ERROR: "
    echo "  Variable DB_USER \"${DB_USER}\" isn't valid ! (Format accepted : [A-Za-z\_\-]+)"
    exit 1
fi
